### DISCUSSION TITLE
Problems with main release

### DISCUSSION THREAD

**Discussion by Roald-Andresen** [joalla/discogs_client#162] [Support]
This is from a crude program.  It is meant to run complete one time only, so it doesn't need to run smoothly or be "perfect".
Extracted and relevant Python code.  Python is **not** my "native" programming language.  I'm using v.2.7.1 of the client API package.

(PyCharm 2024.2.3, Python 3.11.9)

```
all_items = user.collection_folders[0].releases
....
	for item in all_items:
		rel = item.release
		....
		master = rel.master
		.....
		myear = master.main_release.year
```

The master.main_release.year works for most of the releases, but then,  Bham!  I get the error (full trace see below)

I have tried to safe guard for None value of both master, master.main_release and master.main_release.year with try...except, 
```
		if master is not None:
			if master.main_release is not None:
				try:
					if master.main_release.year is not None:
						try:
							myear = master.main_release.year
						except:
							myear = 0
				except:
					myear = 0
```

but the program still crashes.  For all I know, there may be something fishy with the release data in Discogs, but I don't see how I can circumvent that. ???
The strange thing is that when I wrote this code for a trial run a year ago, this was not a problem.

TRACE:
```

Traceback (most recent call last):
  File "/home/roald/projects/python/discogs.py", line 260, in <module>
    run_download()
  File "/home/roald/projects/python/discogs.py", line 153, in run_download
    if master.main_release is not None:
       ^^^^^^^^^^^^^^^^^^^
  File "/home/roald/projects/python/venv/lib/python3.11/site-packages/discogs_client/models.py", line 65, in __get__
    response_dict = instance.fetch(self.name)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/roald/projects/python/venv/lib/python3.11/site-packages/discogs_client/models.py", line 245, in fetch
    self.refresh()
  File "/home/roald/projects/python/venv/lib/python3.11/site-packages/discogs_client/models.py", line 204, in refresh
    data = self.client._get(self.data['resource_url'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/roald/projects/python/venv/lib/python3.11/site-packages/discogs_client/client.py", line 114, in _get
    return self._request('GET', url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/roald/projects/python/venv/lib/python3.11/site-packages/discogs_client/client.py", line 106, in _request
    body = json.loads(content)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by Roald-Andresen** [comment]
Didn't find a solution.  Used the Ruby API instead.

**Comment by JOJ0** [comment]
I think the problem with your error handling is that you try to access `.release`/the release object already, which actually throws the `JSONDecodeError`:

```
if master.main_release is not None:
```

Looking at your example code, I suggest you try it like this:

```
            try:
		if master is not None:
			if master.main_release is not None:
				try:
					if master.main_release.year is not None:
						try:
							myear = master.main_release.year
						except:
							myear = 0
				except:
					myear = 0
            except Exception as err:
                print("Exception:", err)
```

The "inner" try/except can be removed I think.

**Comment by Roald-Andresen** [comment]
This is not my "real" code.  This was just an attempt to narrow down the problem case.  The actual problem still remains, no matter where I handle the exception.  I have tried the various combinations of exception catching, but the problem is still this:  Why do the error occur?
It happens on one specific release, while the release information is actually available.  As mentioned, I rewrote my small application in Ruby, and it ran just fine, returning the release information for all of my collection.

Anyhow, this problem is not of any relevance to me any more.  As mentioned in my comment, I have solved my specific problem by using the Ruby API.

I would have deleted the post, but that seems to not be possible.

**Reply by JOJ0** [reply]
I understand that you moved away from Python for your usecase. This is certainly fine.

Nevertheless it would very much help us, the python3-discogs-client project. If you happen to have a code snippet with catching on the very outside and the traceback that failed please send us!

Also: Please tell us which specific release reproduces the error! I did not find a way to reliably reproduce the error. Thanks in advance, this would be huge help! :-)

**Comment by Roald-Andresen** [comment]
The release is:
https://www.discogs.com/release/11186711-Suzi-Quatro-The-Wild-One-The-Greatest-Hits
The python code I wrote for this have already been deleted
